[browser][tests] Standup System.Net.WebSockets.Client.Tests in CI

kjpou1 opened the pull request:
Part of https://github.com/dotnet/runtime/issues/44666

Outstanding for now are the tests that use LoopBack which depends on websockets and are not supported on browser.  Marked the loopback issues as activeIssue - `[ActiveIssue("#34690", TestPlatforms.Browser)]`

Also, one marked `[ActiveIssue("https://github.com/dotnet/runtime/issues/44720")]` which is being looked into.  https://github.com/dotnet/runtime/issues/44720

kjpou1 committed:
[browser][tests] Standup System.Net.WebSockets.Client.Tests in CI

kjpou1 committed:
Just a small test to see if tests are actually kicked off and failing.

kjpou1 committed:
Remove comment from ActiveIssue to pass tests

lewing reviewed (commented):
Looks ok after the loopback activeissue change

kjpou1 committed:
Merge branch 'master' of https://github.com/dotnet/runtime into wasm-websockets-tests

kjpou1 committed:
Address review comments

- mark the loopback issues as activeIssue - [ActiveIssue("https://github.com/dotnet/runtime/issues/34690", TestPlatforms.Browser)]

kjpou1 commented on src/libraries/System.Net.WebSockets.Client/tests/CloseTest.cs:
```diff
@@ -327,6 +327,7 @@ public async Task CloseAsync_DuringConcurrentReceiveAsync_ExpectedStates(Uri ser
 
         [ConditionalFact(nameof(WebSocketsSupported))]
         [ActiveIssue("https://github.com/dotnet/runtime/issues/34690", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
+        [PlatformSpecific(~TestPlatforms.Browser)] // LoopBack not not supported on browser
```
Addressed in recent commits

lewing commented on src/libraries/System.Net.WebSockets.Client/tests/CloseTest.cs:
```diff
@@ -327,6 +327,7 @@ public async Task CloseAsync_DuringConcurrentReceiveAsync_ExpectedStates(Uri ser
 
         [ConditionalFact(nameof(WebSocketsSupported))]
         [ActiveIssue("https://github.com/dotnet/runtime/issues/34690", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/34690", TestPlatforms.Browser)]
```
These can all be 

```suggestion
        [ActiveIssue("https://github.com/dotnet/runtime/issues/34690", TestPlatforms.Windows | TestPlaforms.Browser, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
        [ActiveIssue("https://github.com/dotnet/runtime/issues/34690", TestPlatforms.Browser)]
```

But I also made an error in my first review comment and meant https://github.com/dotnet/runtime/issues/42852

kjpou1 replied:
Addressed in recent commits

stephentoub reviewed (approved):
Other than the existing comments, LGTM.

kjpou1 committed:
Merge branch 'master' of https://github.com/dotnet/runtime into wasm-websockets-tests

kjpou1 committed:
Address review comments

kjpou1 committed:
Merge branch 'master' of https://github.com/dotnet/runtime into wasm-websockets-tests

lewing committed:
Correct confusion I caused

lewing committed:
Remove extra extra exclusion that snuck in

lewing committed:
Make browser specific failure browser specific

kjpou1 commented:
Thanks for the catch Larry : Make browser specific failure browser specific
